tests/monotouch-test/ObjCRuntime/RuntimeTest.cs

```diff
@@ -489,6 +489,34 @@ public void NSAutoreleasePoolInThreadPool ()
 			obj.Dispose ();
 		}
 
+		[Test]
+		public void NSAutoreleasePoolInThread ()
+		{
+			var count = 10;
+			var threads = new Thread [count];
+			var obj = new NSObject ();
+
+			for (int i = 0; i < count; i++) {
+				threads [i] = new Thread ((v) => {
+					obj.DangerousRetain ().DangerousAutorelease ();
+				}) {
+					Name = $"NSAutoreleasePoolInThread #{i}",
+					IsBackground = true,
+				};
+				threads [i].Start ();
+			}
+
+			for (var i = 0; i < count; i++) {
+				Assert.IsTrue (threads [i].Join (TimeSpan.FromSeconds (1)), $"Thread #{i}");
+			}
+
+			// Strangely enough there seems to be a race condition here, not all threads will necessarily
+			// have completed the autorelease by this point. Some should have though, so assert that the object
+			// was released on at least one thread.
+			Assert.That ((int) obj.RetainCount, Is.LessThan (10), "RC");
```

Comment:
Use `count`, which could be made a `const`, instead of `10`.

> so assert that the object was released on at least one thread.

Just by reading the `NSAutoreleasePoolInThread` I'm not convinced it shows the feature to be working.

E.g. If only done by a single thread (out of 10) then the issue might not be a race. It could be that it somehow get called just  once. 

If you use `count - 1` then we know (but reading the test, not the code) this works on multiple (not just the first) threads.

Reply:
@spouliot 

> Use `count`, which could be made a `const`, instead of `10`.

Done.

> If you use `count - 1` then we know (but reading the test, not the code) this works on multiple (not just the first) threads.

I decided to be braver and used `count / 2` instead :smile: